--- raft/cpp/include/raft/matrix/detail/select_radix.cuh	2023-04-12 07:29:14.000000000 -0700
+++ raft/cpp/include/raft/matrix/detail/select_radix_update.cuh	2023-04-20 19:06:53.323031000 -0700
@@ -110,7 +110,7 @@
   // When writing is not skipped, read `in_buf`(T) and `in_idx_buf`(IdxT), and write `out_buf`(T)
   // and `out_idx_buf`(IdxT).
   // The ratio between these cases determines whether to skip writing and hence the buffer size.
-  constexpr float ratio = 2 + sizeof(IdxT) * 2.0 / sizeof(T);
+  constexpr float ratio = 128;
   return len / ratio;
 }
 
